File: Python/ackermann_function.py
def Ackermann_func(m, n): 

	history = [[0 for i in range(n + 1)] for j in range(m + 1)] 

	for i in range(m + 1): 
		for j in range(n + 1): 

			if i == 0:	 
				history[i][j] = j + 1

			elif j == 0: 
				history[i][j] = history[i-1][1] 

			else: 
				a = i - 1
				b = history[i][j-1] 

				if a == 0:	 
					result = b + 1
				elif b <= n: 
					result = history[i-1][history[i][j-1]] 
				else: 
					result = (b-n)*(a) + history[a][n] 

				history[i][j] = result 

	return history[m][n] 

m, n = input("Enter non-negative values of m and n: ").split()
m, n = int(m), int(n)

print("A(%u,%u) = %u" % (m,n,Ackermann_func(m, n)))


#Time complexity: O( M * N )
#Space complexity: O( M * N )

File: PHP/math/ackermann_function.php
<!-- Ackermann function is one of the simplest examples of a total computable function,
    that is not primitive recursive.
    It’s a function with two arguments,
    each of which can be assigned any non-negative integer. -->

<?php

function ackermann( $m , $n )
{
    if ( $m==0 )
        return $n + 1;
    elseif ( $n==0 )
        return ackermann( $m-1 , 1 );
    return ackermann( $m-1, ackermann( $m , $n-1 ) );
}

echo "The result is : " .ackermann( 2, 3 );

?>

<!-- Output Screen =>

The result is : 9

Time complexity : O(mA(m, n))
Space complexity : O(m) -->
